test/commands.test.ts

```diff
@@ -0,0 +1,259 @@
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            select(e, textWordRange);
+            await vscode.commands.executeCommand('org.verbose');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('Literal', async () => {
+        const initial = 'Some text here';
+        const expected = ': Some text here';
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            await vscode.commands.executeCommand('org.literal');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('InsertSubheading', async () => {
+        const initial = `* Header
+* Header2`;
+        const expected = `* Header\n` +
+            `** \n` +
+            `*** \n` +
+            `* Header2`;
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            await vscode.commands.executeCommand('org.insertSubheading');
+            await vscode.commands.executeCommand('org.insertSubheading');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('IncrementContext', async () => {
+        const steps = [
+            '* Header',
+            '* TODO Header',
+            '* DONE Header',
+        ];
+
+        await inTextEditor({ language: 'org', content: steps[0] }, async (e, d) => {
+            for (let i = 1; i < steps.length; ++i) {
+                await vscode.commands.executeCommand('org.incrementContext');
+                assert.equal(d.getText(), steps[i]);
+            }
+        });
+    });
+
+    test('DecrementContext', async () => {
+        const steps = [
+            '* DONE Header',
+            '* TODO Header',
+            '* Header',
+        ];
+
+        await inTextEditor({ language: 'org', content: steps[0] }, async (e, d) => {
+            for (let i = 1; i < steps.length; ++i) {
+                await vscode.commands.executeCommand('org.decrementContext');
+                assert.equal(d.getText(), steps[i]);
+            }
+        });
+    });
+
+    test('DemoteSubtree', async () => {
+        const initial = `* Header 1
+** Subheader
+* Header 2`;
+
+        const expected = `** Header 1
+*** Subheader
+* Header 2`;
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            await vscode.commands.executeCommand('org.demoteSubtree');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('PromoteSubtree', async () => {
+        const initial = `** Header 1
+*** Subheader
+** Header 2`;
+
+        const expected = `* Header 1
+** Subheader
+** Header 2`;
+
+        await inTextEditor({ language: 'org', content: initial }, async (e, d) => {
+            await vscode.commands.executeCommand('org.promoteSubtree');
+            assert.equal(d.getText(), expected);
+        });
+    });
+
+    test('Timestamp', async () => {
```

Comment:
This test is failing for me (4 last tests are failing, actually) because the code wraps timestamp in square brackets while this expects angled brackets.

Reply:
Yep, that's expected. Tests are failing because expected result was taken from Emacs and vscode-org-mode and emacs' org-mode behave differently. So, I assume that this is a bug that need to be fixed. I raised an issue for promote/demote subtree, but didn't yet raised issue for timestamp command and 'insert heading respecting content' command.

Reply:
See my other comment about backquoted multi-line strings in the PR changed files.  I also added an issue regarding timestamp command with proposed solution.  I'll probably pick it up to work on next after checkboxes get approved.